sky/packages/sky/lib/src/widgets/scrollable.dart

```diff
@@ -379,6 +379,34 @@ class Block extends StatelessComponent {
   }
 }
 
+abstract class ScrollingListPainter extends Painter {
+  RenderBlockViewport get renderer => renderObject as RenderBlockViewport;
+
+  bool get isVertical => renderer.isVertical;
+
+  Size get viewportSize => renderer.size;
+
+  double get contentExtent {
+    return isVertical
+      ? renderer.getMaxIntrinsicHeight(const BoxConstraints())
+      : renderer.getMaxIntrinsicWidth(const BoxConstraints());
```

Comment:
These calls are super expensive.  Can't we get this information from the scroll behavior?